File: src/test/data-structures-and-algorithms/primitive-types/counting-bits-test.cc
#include "src/main/data-structures-and-algorithms/primitive-types/counting-bits.h"

#include "gtest/gtest.h"

#include <vector>

TEST(CountingBits, FirstExample) {
  int n{2};
  epi::CountingBits algo;
  std::vector<int> expected{0,1,1};
  ASSERT_EQ(expected, algo.countBits(n));
}

TEST(CountingBits, DISABLED_SecondExample) {
  int n{5};
  epi::CountingBits algo;
  std::vector<int> expected{0,1,1,2,1,2};
  ASSERT_EQ(expected, algo.countBits(n));
}

File: src/test/clj/data-structures-and-algorithms/primitive-types/counting-bits-test.clj
(ns tests.counting-bits-test 
  (:require [data-structures-and-algorithms.primitive-types.counting-bits :as algo])
  (:use clojure.test))

(deftest must-pass (is true))

(deftest zeroth-example 
  (is (= (algo/count-bits 0) 0)))

(deftest first-example
  (is (= (algo/count-bits 1) 1)))

;(deftest second-example
;  (is (= (algo/count-bits 2) 1)))

;(deftest third-example
;  (is (= (algo/count-bits 3) 2)))
